test/inference/hmc.jl

```diff
@@ -223,4 +223,21 @@
         chain = sample(gdemo_default_prior, alg, 10_000)
         check_numerical(chain, [:s, :m], [mean(InverseGamma(2, 3)), 0], atol=0.3)
     end
+
+    @turing_testset "warning for difficult init params" begin
+        attempt = 0
+        @model function demo_warn_init_params()
+            x ~ Normal()
+            if (attempt += 1) < 30
+                Turing.@addlogprob! -Inf
+            end
+        end
+
+        @test_logs (
+            :warn,
+            "failed to find valid initial parameters in 10 tries; consider providing explicit initial parameters using the `init_params` keyword",
+        ) min_level = Logging.Warn begin
```

Comment:
Apparently Logging is not defined. But based on the docs, I think you could avoid it by using
```suggestion
        ) (:info,) match_mode=:any begin
```